### PR TITLE
Merge managers into one object

### PR DESCRIPTION
**Pull request by jrg94**
Made some massive changes to the way the game operates under the hood. No surface level changes were made.

Anything that is not the game start script (excluding camera, light, and EventSystem) are no spawned into the scene via a script. All managers have either moved to the managers prefab or have been deemed not actual managers (i.e. See ResourceIncrementer). 

Additional work will need to be made to further clean up the game from under the hood. This is just the beginning of a large unhackathon.

### CODE REVIEW

File: client/poof_game/assets/Scripts/src/UI/TestModelPanel.cs [hunk 1]
```diff
@@ -3,26 +3,25 @@
...
-	
 
 	void Awake(){
-		modalPanel = ModalPanel.Instance ();
 		displayManager = DisplayManager.Instance ();
 		myButton1Action = new UnityAction (TestButton1Function);
 		myButton2Action = new UnityAction (TestButton2Function);
-		myButton3Action = new UnityAction (TestButton3Function); 
+		myButton3Action = new UnityAction (TestButton3Function);
+        myCancelButtonAction = new UnityAction(TestCancelFunction);
```
Comment (CatalystOfNostalgia) [inline client/poof_game/assets/Scripts/src/UI/TestModelPanel.cs:20]: Tab this over.

File: client/poof_game/assets/Scripts/src/UI/SettingsMenu.cs [hunk 2]
```diff
@@ -15,40 +15,51 @@ public class SettingsMenu : MonoBehaviour {
...
+        masterVolumeSlider.onValueChanged.AddListener(delegate { SoundManager.soundManager.masterVolume = masterVolumeSlider.value; });
+
+        musicVolumeSlider.onValueChanged.RemoveAllListeners();
+        musicVolumeSlider.onValueChanged.AddListener(delegate { SoundManager.soundManager.musicVolume = musicVolumeSlider.value; });
+
+        soundVolumeSlider.onValueChanged.RemoveAllListeners();
+        soundVolumeSlider.onValueChanged.AddListener(delegate { SoundManager.soundManager.soundVolume = soundVolumeSlider.value; });
+    }
+
+    void ClosePanel(){
```
Comment (CatalystOfNostalgia) [inline client/poof_game/assets/Scripts/src/UI/SettingsMenu.cs:62]: There's some inconsistent tabbing here.

**Comment by jrg94** [comment]
Should be sorted now

File: client/poof_game/assets/Scripts/src/UI/SettingsMenu.cs (hunk 2 above)
Comment (jrg94) [inline client/poof_game/assets/Scripts/src/UI/SettingsMenu.cs:62]: Should be sorted now.

**Pull request merged**